crates/symbolicator/src/services/shared_cache.rs

```diff
@@ -207,7 +200,50 @@ impl GcsState {
         }
     }
 
-    async fn store(&self, key: SharedCacheKey, mut src: File) -> Result<SharedCacheStoreResult> {
+    async fn exists(&self, key: &SharedCacheKey) -> Result<bool> {
+        let token = self.get_token().await?;
+        let url = gcs::object_url(&self.config.bucket, key.gcs_bucket_key().as_ref())?;
+        let request = self.client.get(url).bearer_auth(token.as_str()).send();
+        let request = tokio::time::timeout(CONNECT_TIMEOUT, request);
+
+        match request.await {
+            Ok(Ok(response)) => match response.status() {
+                StatusCode::OK => Ok(true),
+                StatusCode::NOT_FOUND => Ok(false),
+                status => Err(anyhow!("Unexpected status code from GCS: {}", status)),
```

Comment:
Hum, re-reading this I realise I never fetch the entire response body: the response body is only guaranteed to be received after calling something like `request.bytes()`, `.string()` or `.json()` I think.  This response body is a little bit of JSON data, probably so small it is already received in the initial packet.

But still, should you always try and consume the response or does it not matter?